classes/controlledVocab/ControlledVocab.php

```diff
@@ -1,108 +1,146 @@
 <?php
-/**
- * @defgroup controlled_vocab Controlled Vocabulary
- */
 
 /**
  * @file classes/controlledVocab/ControlledVocab.php
  *
- * Copyright (c) 2014-2021 Simon Fraser University
- * Copyright (c) 2000-2021 John Willinsky
+ * Copyright (c) 2024 Simon Fraser University
+ * Copyright (c) 2024 John Willinsky
  * Distributed under the GNU GPL v3. For full terms see the file docs/COPYING.
  *
  * @class ControlledVocab
  *
- * @ingroup controlled_vocab
- *
- * @see ControlledVocabDAO
- *
- * @brief Basic class describing an controlled vocab.
+ * @brief ControlledVocab model class
  */
 
 namespace PKP\controlledVocab;
 
-use PKP\db\DAORegistry;
+use Eloquence\Behaviours\HasCamelCasing;
+use Illuminate\Database\Query\JoinClause;
+use Illuminate\Support\Facades\DB;
+use Illuminate\Database\Eloquent\Model;
+use Illuminate\Database\Eloquent\Casts\Attribute;
+use Illuminate\Database\Eloquent\Builder;
+use PKP\facades\Locale;
 
-class ControlledVocab extends \PKP\core\DataObject
+class ControlledVocab extends Model
 {
-    //
-    // Get/set methods
-    //
+    use HasCamelCasing;
+
+    protected $table = 'controlled_vocabs';
+    protected $primaryKey = 'controlled_vocab_id';
+
+    protected $guarded = [
+        'controlled_vocab_id',
+    ];
 
     /**
-     * get assoc id
+     * Indicates if the model should be timestamped.
      *
-     * @return int
+     * @var bool
      */
-    public function getAssocId()
+    public $timestamps = false;
```

Comment:
Wondering is there is specific reason we are not adding timestamp columns to the table when we are updating those to use the eloquent model ? It does not have any effect right away but in long run we may start using those .

Reply:
I have no objection to adding these -- but as you say, we're not using them yet, so it hasn't been a priority.